scripts/repoint_frontend.sh

```diff
@@ -0,0 +1,272 @@
+#!/bin/bash
+
+: '
+Usage:
+scripts/repoint_frontend.sh -h
+scripts/repoint_frontend.sh --help 
+scripts/repoint_frontend.sh --list-steps
+scripts/repoint_frontend.sh 0000 gith1
+scripts/repoint_frontend.sh 0000 gith1 --from-step 1
+scripts/repoint_frontend.sh 0000 gith1 --only-step 1
+
+When this script is modified the following steps should be taken:
+- Check the script works in standard mode              scripts/repoint_frontend.sh 0000 gith1
+- Check step listing works                             scripts/repoint_frontend.sh --list-steps
+- Check starting from a step works                     scripts/repoint_frontend.sh 0000 gith1 --from-step 1
+- Check running a step in isolation works              scripts/repoint_frontend.sh --only-step 2
+- Check directory requirement is checked and works     cd scripts && scripts/repoint_frontend.sh --list-steps
+- Check AWS login status check works                   aws sso logout && scripts/repoint_frontend.sh
+- Check error is returned on too many positional args  scripts/repoint_frontend.sh 0000 gith1 invalid
+- 
+'
+
+true=0
+false=1
+
+# $1: message
+info() {
+  echo -e "[ INFO  ]: $1"
+}
+
+# $1: message
+info_overwrite() {
+  echo -n -e "\r[ INFO  ]: $1"
+}
+
+# $1: lines
+info_multiline() {
+  while IFS= read -r line; do
+    info "$line"
+  done <<< "$1"
+}
+
+# $1: message
+step() {
+  echo -e "[ STEP  ]: $1"
+}
+
+# $1: message
+query() {
+  read -p "[ QUERY ]: $1: " response
+}
+
+# $1: message, $2: info
+error() {
+  echo -e "[ ERROR ]: $1"
+
+  if [[ -n "$2" ]]; then
+    info "$2"
+  fi
+
+  exit 1
+}
+
+# $1: message
+warn() {
+  echo -e "[ WARN  ]: $1"
+}
+
+spacer() {
+  echo ""
+}
+
+# $1: message, $2: info
+exit_on_failure() {
+  if [[ $? -ne 0 ]]; then
+    error "$1" "$2"
+  fi
+}
+
+# check if in project root
+current_dir=$(basename "$PWD")
+if [[ "$current_dir" != "communications-manager-api" ]]; then
+  error "you must be in the project root directory to run this script"
+fi
+
+# check aws login status
+aws sts get-caller-identity > /dev/null 2>&1
+exit_on_failure "you must have an active AWS SSO session to run this script"
+
+positional_arg_index=0
+
+# shifts positional args from arguments into named vars
+handle_positional_arg() {
+  case $((positional_arg_index++)) in
+    0) ticket_id="$1";;
+    1) shortcode="$1";;
+    *) error "too many positional arguments"
+  esac
+}
+
+# FROM_STEP: where to continue execution from, used to skips steps already run
+# 0: remove mTLS (if set)
+# 1: create new branch for proxy modifications
+# 2: modify proxy files
+# 3: stage, commit, and push changes
+# 4: await domainName available status (or timeout)
+
+from_step=0
```

Comment:
Instead of having this part where you skip steps, could you not just check if mTLS is enabled and skip if it isn't. The ability to skip steps seems like an unnecessary complexity?

Reply:
Why do you need it to create a branch for you? Could the tool just not work from a branch

Reply:
Because I'm just that lazy XD it also automatically commits to the branch and pushes

Reply:
Commiting and pushing seems fine but yeah, the ability of skipping steps etc makes this CLI script seem a bit overcomplicated and probably one for just manual intervention